dropwizard-hibernate/src/main/java/io/dropwizard/hibernate/UnitOfWorkAwareProxyFactory.java

```diff
@@ -122,4 +111,39 @@ public UnitOfWorkAspect newAspect() {
     public UnitOfWorkAspect newAspect(Map<String, SessionFactory> sessionFactories) {
         return new UnitOfWorkAspect(sessionFactories);
     }
+
+    class MethodInterceptor {
+
+        private final Map<String, SessionFactory> sessionFactories;
+
+        public MethodInterceptor(Map<String, SessionFactory> sessionFactories) {
+          this.sessionFactories = sessionFactories;
+        }
+
+        @RuntimeType
+        Object invoke(@This Object self, @Origin Method overridden, @SuperMethod Method proceed,
```

Comment:
Byte-buddy also defines the `@SuperCall` annotation to use a parameter of type `@SuperCall Callable<?>`. As we use the `@This` and `@AllArguments` annotated parameters only for the invocation of the super method, maybe we could simplify this here.

Reply:
I was trying to emulate the existing code but that's definitely cleaner.  I've revised the PR.